### ISSUE TITLE
ARTIK IDE Installer - Upgrade Function

### ISSUE THREAD

**Issue by bmicklea** [codenvy/artik-ide#285]
The ARTIK IDE installer should allow users to upgrade their version of ARTIK. This will pull down a new installer bundle and then upgrade all aspects (Docker if needed), Electron client and ARTIK Server.

**Comment by eivantsov** [comment]
@bmicklea for now we do not have any mechanism to check for a new version, so expect users just to download a new installer.

**Comment by bmicklea** [comment]
@slemeur I believe in Che there is a plan to create notifications when new versions are pushed to DockerHub correct?  If that's the case then this can be done after that capability is added. In the meantime as Eugene said, we need to clarify the release information channel for ARTIK users and suggest that they download a new installer when a new version is available.

**Comment by slemeur** [comment]
@eivantsov: There is no mechanisme at this moment, but why can't the installer check the latest versions pushed on dockerhub?

@bmicklea : Yes we are thinking about it for Che. But no timeframe at this moment.

**Comment by TylerJewell** [comment]
Notifications on new versions should be out of scope.  The goal was entirely offline operation initially. We don't have a notification system.  Though the Che cli lets you check dockerhub for a list of versions though I think artik is looking at wrong URL.

**Comment by bmicklea** [comment]
Correct, for the GA version and for some time after to upgrade users will:
- Download a new installer
- Run the installer

This will replace their existing ARTIK IDE with a new version.

**Comment by eivantsov** [comment]
@slemeur you are talking about Artik IDE upgrade. While I am talking about both Artik IDE upgrade and the Electron client upgrade.
